File: Data Structures & Algorithms/two-integer-sum/submission-1.go
func twoSum(nums []int, target int) []int {
    positions := map[int]int{}
	for i, n := range nums {		
		if j, seen := positions[target - n]; seen {
			return []int{j, i}
		}
		positions[n] = i
	}
	return []int{}
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while True:
            m = int((l + r) / 2)
            if nums[m] == target:
                return m
            elif nums[r] == target:
                return r
            elif l == m:
                return -1
            elif nums[m] < target:
                l = m
            else:
                r = m